src/coreclr/vm/object.cpp

```diff
@@ -1473,62 +1473,38 @@ void StackTraceArray::CheckState() const
         assert(p[i].pFunc != NULL);
 }
 
-void StackTraceArray::Grow(size_t grow_size)
+void StackTraceArray::Allocate(size_t size)
 {
     CONTRACTL
     {
         THROWS;
         GC_TRIGGERS;
         MODE_COOPERATIVE;
-        INJECT_FAULT(ThrowOutOfMemory(););
         PRECONDITION(IsProtectedByGCFrame((OBJECTREF*)this));
     }
     CONTRACTL_END;
 
-    size_t raw_size = grow_size * sizeof(StackTraceElement) + sizeof(ArrayHeader);
+    size_t raw_size = size * sizeof(StackTraceElement) + sizeof(ArrayHeader);
```

Comment:
This multiplication can overflow

Reply:
Hmm, right, on 32 bit OSes. I haven't noticed there was this preexisting issue, thank you for noticing it.

Reply:
Fixed